Quick start guide (#42)
inngest/agent-kit: docs/getting-started/quick-start.mdx

---
title: Quick start
description: "Learn the basics of AgentKit in a few minutes."
---

> In this tutorial you will create an [Agent](/concepts/agents) and a [Network](/concepts/networks) with AgentKit and run it using the Inngest Dev Server.

Reviewer: ```suggestion
In this tutorial you will create an [Agent](/concepts/agents) and run it within a [Network](/concepts/networks) with AgentKit.
```

    You'll also specify which `model` you want the agent to use. Here we'll use Anthropic's [Claude 3.5 Haiku](https://docs.anthropic.com/en/docs/about-claude/models) model. ([Model reference](/concepts/models))

    Your agent can be whatever you want, but in this quick start, we'll create a Postgres database administrator agent:

    ```ts index.ts
    import { createAgent, anthropic } from '@inngest/agent-kit';

    const dbaAgent = createAgent({
      name: 'Database administrator',
      description: 'Provides expert support for managing PostgreSQL databases',
      system: 'You are a PostgreSQL expert database administrator. ' +
        'You help answer questions about Postgres including database schema, indexes, extensions.',
      model: anthropic({
        model: 'claude-3-5-haiku-latest',
        max_tokens: 1000,
      }),
    });
    ```

    We'll also have to set any provider API keys as environment variables.

    ```shell terminal
    export ANTHROPIC_API_KEY=sk-ant-api03-XXXXXX....
    ```

  </Step>
  <Step title="Run the server">

>     Next, we'll create an http server in order to run our agent. In the same

Reviewer: ```suggestion
    Next, we'll create an http server in order to run our agent. In the same file as our _Agent_ definition:
```

    ```shell terminal
    npx tsx ./index.ts
    ```

  </Step>

  <Step title="Test our agent">
    To test our agent, we'll use the [Inngest dev server](https://www.inngest.com/docs/local-development) to allow us to visually debug our agents. Using `npx`, we'll start the server and point it to our AgentKit server:

    ```shell terminal
    npx inngest-cli@latest dev -u http://localhost:3000/api/inngest
    ```

    Now, open the dev server and select the function's tab (`http://localhost:8288/functions`) and click the "Invoke" button:

    ![Inngest Dev Server function list](/graphics/quick-start/dev-server-agent.png)

    In the Invoke function modal, specify the input prompt for your agent and click the "Invoke function" button.

    ![Inngest Dev Server invoke function modal](/graphics/quick-start/dev-server-invoke.png)

    ```json Invoke payload
    {
      "data": {
        "input": "How do I aggregate an integer column across a date column by week?"
      }
    }
    ```

    You'll be redirected to watch the agent run and view the output:

    ![Inngest Dev Server agent run](/graphics/quick-start/dev-server-agent-run.png)

  </Step>

> <Step title="Creating a Network">

Reviewer: Explaining that a Network is useful to have multiple Agents interact and then showcase a Network with a single Agent is weird.
We could:
1. Remove the Network from the Quickstart
2. Add another simple Agent to the mix

I'll prefer option 2.
